eng/GenAPI.exclusions.txt

```diff
@@ -4,6 +4,8 @@ T:Microsoft.AspNetCore.Components.RenderTree.RenderTreeFrame
 T:Microsoft.AspNetCore.Mvc.ApplicationModels.PageParameterModel
 T:Microsoft.AspNetCore.Mvc.ApplicationModels.PagePropertyModel
 # Manually implemented - https://github.com/aspnet/AspNetCore/issues/8825
+T:Microsoft.AspNetCore.Components.AuthorizeView
+T:Microsoft.AspNetCore.Components.CascadingAuthenticationState
```

Comment:
Just curious: Are these additions and the related movement between ref assemblies about catching up with previous changes made in product assemblies?

Reply:
This was left over from #10227. We have to write out the ref assembly sources for these components manually because the generator doesn't retain private setters. They have to be retained otherwise the tooling won't work.

Reply:
Which tooling? Is there an issue tracking this? I'm not familiar with this problem, but I find it surprising that we need private members in a ref assembly. So far, we've really only meant for ref assemblies to represent public API.

Reply:
The Razor editor shipped in VS needs to identify component parameters, and one of the conditions it's looking for is that there's a setter (regardless of visibility). If the ref assemblies claim it's a true read-only property, then the parameters don't show up in intellisense.

The tracking issue is the one mentioned on line 6, https://github.com/aspnet/AspNetCore/issues/8825

Longer term, maybe we could relax the requirement for there to be a setter, and just look for the `[Parameter]` attribute. We could also have an analyzer that warns people not to create `[Parameter]` properties that are truly read-only. @rynowak is the expert for this - what's your expectation of the long-term solution, Ryan?

Reply:
Actually, for any follow-up discussion on this topic, please use #8825 :)

Marking as resolved.

Reply:
My preference would be that we make the properties public and change the analyzer to flag *property set* operations.

We've got workarounds in 2-3 different systems now to support the properties being non-public.